### PR TITLE
[Megafauna] Add weakpoints and anatomical proficiencies

### LINKED ISSUE #62491: [MegaFauna] Add Relevant Proficiencies to Megafauna
### Is your feature request related to a problem? Please describe.

The various wildlife in the megafauna mod lack the 'Principles of Biology' Proficiency, which is a stark contrast to the basic wildlife in CDDA.

### Solution you would like.

Add the Principles of Biology Proficiency to the Megafauna weakpoint list.

### Describe alternatives you have considered.

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by merrygin**
#### Summary
Mods "Add weakpoints system to Megafauna mod"

#### Purpose of change
Fixes #62491
Megfauna creatures do not have anatomical proficiencies nor weakpoints that make those useful yet. Purpose is rectifying that.

#### Describe the solution

- [x] Add families "simple biology"
- [x] Add weakpoints (default stuff for the smaller creatures)
- [x] Add "megfaunal biology" family (?)
- [x] Add weakpoints (new and specific to actual megafauna)

#### Describe alternatives you've considered
Just simple biology and copy and paste weakpoints just to have something quick, but I'd like to take the time and do it properly.

#### Testing
tbd

#### Additional context
Tweaked (buffed, mostly) most armor values as well to make WPs more important.

### CODE REVIEW

File: data/mods/Megafauna/monsters/mf_domestic.json [hunk 1]
```diff
@@ -148,7 +152,8 @@
     "default_faction": "mega_herbivore",
     "bodytype": "horse",
     "categories": [ "WILDLIFE" ],
-    "species": [ "MAMMAL" ],
+    "species": [ "MAMMAL" ],
```
[the remark this answers is not in the document]
Comment (merrygin) [reply data/mods/Megafauna/monsters/mf_domestic.json:155]: who's a good boi!?

File: data/mods/Megafauna/mf_weakpoint_sets.json [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        "difficulty": { "melee": 2, "ranged": 4 },
+        "coverage_mult": { "ranged": 0.5 },
+        "coverage": 25,
+        "required_effects": [ "downed" ],
+        "effects": [
+          {
+            "effect": "nausea",
+            "permanent": true,
+            "chance": 10,
+            "message": "You have ripped the %s bowels!  The % is wretching.",
```
Comment (merrygin) [inline data/mods/Megafauna/mf_weakpoint_sets.json:88]:
```suggestion
            "message": "You have ripped the %s bowels!  The % is retching.",
```

**Comment by merrygin** [comment]
@LyleSY , sorry for pinging, but would you terribly mind to have a look if this looks sensible? I mainly oriented myself on the dino weakpoints, but added some stuff (like striking vitals, easier targetting of downed animals) and I'm not sure if the values are sensible like that.

**Review by LyleSY — APPROVED** [review]
All of the logic and the specific values here make sense to me. Playtesting may show some things I can't see, but this all looks great to me

**Comment by merrygin** [comment]
> All of the logic and the specific values here make sense to me. Playtesting may show some things I can't see, but this all looks great to me

Thanks a bunch!

File: data/mods/Megafauna/proficiencies/mf_weakpoints.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    "id": "prof_wp_megafauna",
+    "category": "prof_weakpoint",
+    "name": { "str": "Megafaunal Biology" },
+    "description": "You are familiar with the general anatomy of megafauna, and can exploit potential weaknesses.",
+    "required_proficiencies": [ "prof_intro_biology" ],
+    "can_learn": true,
+    "time_to_learn": "6 h",
+    "default_weakpoint_bonus": 4,
+    "default_weakpoint_penalty": -2
+  }
```
Comment (Maleclypse) [inline data/mods/Megafauna/proficiencies/mf_weakpoints.json:13]:
```suggestion
  } 
```

File: data/mods/Megafauna/proficiencies/mf_weakpoints.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
```
Comment (Maleclypse) [inline data/mods/Megafauna/proficiencies/mf_weakpoints.json:13]:
```suggestion
  }
```

File: data/mods/Megafauna/mf_weakpoint_sets.json [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        ]
+      },
+      {
+        "name": "the eyes",
+        "//": "standing animal, eyes are nigh impossible to hit (with non point weapons.  actually, reach weapons, but not possible here, so point it is)",
+        "armor_mult": { "physical": 0 },
+        "coverage": 1,
+        "crit_mult": { "all": 1.25 },
+        "coverage_mult": { "ranged": 0.5 },
+        "difficulty": { "ranged": 6, "melee": 10, "point": 6 },
```
Comment (ZhilkinSerg) [inline data/mods/Megafauna/mf_weakpoint_sets.json:126]: As far as I can understand, if weakpoint contains `melee` it cannot contain `point`:

https://github.com/CleverRaven/Cataclysm-DDA/blob/e27cd618d85993af272d34458085aa2ef95b2045/src/weakpoint.cpp#L234

Comment (Maleclypse) [inline data/mods/Megafauna/mf_weakpoint_sets.json:126]:
```suggestion
        "difficulty": { "ranged": 6, "melee": 10 },
```

**Pull request merged**